packages/react-instantsearch-core/src/connectors/__tests__/connectQueryRules.ts

```diff
@@ -326,6 +326,14 @@ describe('connectQueryRules', () => {
       });
 
       it('slices and warns in development when more than 10 rule contexts are applied', () => {
+        // We need to simulate being in development mode and to mock the global console object
+        // in this test to assert that development warnings are displayed correctly.
+        const originalNodeEnv = process.env.NODE_ENV;
+        process.env.NODE_ENV = 'development';
+        const originalConsole = global.console;
+        const warnSpy = jest.fn();
```

Comment:
You removed `spyOn` it wasn't working?

Reply:
`spyOn` works but still displays the message in the console. This solution doesn't pollute the console as much.

Reply:
```
jest.spyOn(object, methodName).mockImplementation(() => customImplementation)
```

> https://jestjs.io/docs/en/jest-object#jestspyonobject-methodname

Reply:
Nice, thanks!